include/time.h

```diff
@@ -217,7 +217,7 @@ time_t time(FAR time_t *timep);
 
 #ifdef CONFIG_HAVE_DOUBLE
 double difftime(time_t time1, time_t time0);
-#else
+#elif defined(CONFIG_HAVE_FLOAT)
 float difftime(time_t time1, time_t time0);
```

Comment:
Shouldn't have an "#else" with #warn to let user know difftime() is not supported?

Reply:
linker will report the error

Reply:
Maybe we need to guide the user to find the right solution, compilation errors are not helpful

Reply:
but user mayn't call difftime at all, it's bad to warn he without any condition. BTW, there are many prototypes are guided by `#ifdef CONFIG_xxxx`, do we need any warning to all places with `#else`.

Reply:
We probably should have a catch-all `else`, or strange problems will result!

Reply:
what's strange problem? @hartmannathan . It's verry a common pattern in NuttX code base to skip the function prototype and implementation if the requirement doesn't meet,

Reply:
> what's strange problem? @hartmannathan . It's verry a common pattern in NuttX code base to skip the function prototype and implementation if the requirement doesn't meet,

@xiaoxiang781216 I meant that no prototype would exist, so compilation of other files would fail with error or warning. But on second thought, I think this is okay.

Reply:
Ok, let's merge this patch, thanks.